docking.js

```diff
@@ -405,7 +405,7 @@ var DockedDash = GObject.registerClass({
         if (Main.uiGroup.contains(global.top_window_group))
             Main.uiGroup.set_child_below_sibling(this, global.top_window_group);
 
-        if (settings.get_boolean('dock-fixed')) {
+        if (settings.get_boolean('dock-fixed') && !settings.get_boolean('manualhide')) {
```

Comment:
This keeps the user's visibility setting, it will just prevent the dock from being visible initially if `manualhide` is active.

This way we don't need to keep track of the visibility setting ourselves when the dock is disabled.